keps/sig-scheduling/4247-queueinghint/README.md

```diff
@@ -0,0 +1,1122 @@
+  - Mitigations: The scheduling queue priodically flushing Pods in the unschedulable Pod pool. So, even if such bug exists, Pods' scheduling are retried after a certain period, which is 5 min by default. You can shorten the max duration that Pods can stay in the unschedulable Pod pool by using `--pod-max-in-unschedulable-pods-duration`.
+  - Diagnostics: If you increases the log level to more than 5, you can see the logs related to `QueueingHint` in the scheduling queue. Also, the in-tree plugins emits all logs in QueueingHint with log level 5. (If you have a custom plugin, you may want to check the log level in its QueueingHint.)
+  - Testing: There are multiple unit tests to confirm `flushUnschedulablePodsLeftover` is working expectedly.
+
+###### What steps should be taken if SLOs are not being met to determine the problem?
+
+## Implementation History
+
+<!--
+Major milestones in the lifecycle of a KEP should be tracked in this section.
+Major milestones might include:
+- the `Summary` and `Motivation` sections being merged, signaling SIG acceptance
+- the `Proposal` section being merged, signaling agreement on a proposed design
+- the date implementation started
+- the first Kubernetes release where an initial version of the KEP was available
+- the version of Kubernetes where the KEP graduated to general availability
+- when the KEP was retired or superseded
+-->
+
+- Jun 26, 2023: The QueueingHint is implemented and the `EnqueueExtension` interface is changed.
+- Jul 15, 2023: The feature gate is implemented. (enabled by default)
+- Jul 18, 2023: The scheduling queue tracks the Pod being processed to put it back to an appropriate queue.
+- Oct 01, 2023: The initial KEP is submitted.
+
+## Drawbacks
+
+<!--
+Why should this KEP _not_ be implemented?
+-->
+
+## Alternatives
+
+<!--
+What other approaches did you consider, and why did you rule them out? These do
+not need to be as detailed as the proposal, but should include enough
+information to express the idea and why it was not acceptable.
+-->
+
+### Return `QueueImmediately`, `QueueAfterBackoff`, and `QueueSkip` from `QueueingHintFn` instead of introducing new status `Pending`
+
+Instead of requeueing Pods based on why it was rejected, we can do the same by introducing separate `QueueingHint` for queueing - `QueueImmediately` and `QueueAfterBackoff`.
+
+But, as explained in [When to skip/not skip backoff](#when-to-skipnot-skip-backoff), the backoff is a penalty of wasting the scheduling cycle. Also, some few scenario (DRA) don't waste the scheduling cycle, they reject Pods in that scheduling cycle though.
+
+So, whether skipping backoff or not, it's something very close to why the Pod was rejected,
+and thus it's easier to be decided when the Pod is rejected than when the Pod is actually requeued.
+
+### Implement `Blocked` status to block a next scheduling retry until the plugin returns `Queue`
```

Comment:
On second thought, we don't need `Blocked` (at least for the current in-tree plugins), give the reason described here.
Will update https://github.com/kubernetes/kubernetes/pull/119517 once this PR reaches an agreement.